packages/react-ui/src/widgets/BarWidgetUI/BarWidgetUI.js

```diff
@@ -44,13 +45,16 @@ function BarWidgetUI(props) {
     height,
     filterable,
     animation,
-    isLoading
+    isLoading,
+    locale
```

Comment:
If you use the UI it could be useful to keep locale prop.
Perhaps I need more context from @Josmorsot approach to apply with locale, redux. 
¿Do you have an example using setClient?

Reply:
If we want to have strict control over locale, yes, we need to pass it somehow using react mechanism.

`client` approach is just a global variable, that doesn't have responsivness.

Now, `dispatch(setLocale())` will cause proper rerender of affected components. Pure `setLocale` (global setter) will not, unless we implement complex, unnecessary notification mechanism ... aka like react context.

I already proposed to use `IntlContext` [here](https://github.com/CartoDB/carto-react/pull/727#pullrequestreview-1512604592) to get locale provided by `ReactIntlProvider` and ... fallback to default if it's missing. This API would be most robust and would be compatible with "react-intl" way of doing things.

Now, in this situation we can only get `locale` from externally provided object like this:

```
const useOurIntl = () => {
  const externalIntl = useContext(IntlContext)
  const locale = externalIntl?.locale || DEFAULT_LOCALE
  const messagesLocale = findMatchingMessagesLocale(locale)
  return createIntl({locale: externalIntl?.locale, messages: messages[messagesLocale]) // ofc memo and cache and stuff
}
```

now, if app doesn't have ReactIntlProvider, c4r would fallback to `en-EN`
else we would use what user has specified in app

my subjective evaluation
 - no redux,
 - no prop drilling
 - responsive
 - simple enough imo
 - ReactIntl-way

@VictorVelarde @menusal

Reply:
> `fingMatchingMessagesLocale`

there is not-yet-standard api for this:
 * https://github.com/tc39/proposal-intl-localematcher
 * poly/ponyfill: https://github.com/tc39/proposal-intl-localematcher

```
> const { match } = require('@formatjs/intl-localematcher')
> match(['en'], ['es-ES', 'es-MX', , 'en-US'], 'en')
'en-US'
> match(['es'], ['es-ES', 'es-MX', 'en-US'], 'en')
'es-ES'
> match(['es-MX'], ['es-ES', 'es-MX', 'en-US'], 'en')
'es-MX'
> match(['es-XX'], ['es-ES', 'es-MX', 'en-US'], 'en')
'es-ES'
```

Reply:
To me one key point is avoid prop drilling.
I think zbigg's proposal is the simplest and keeps that in control, and aligned to the library model of work. This would be my vote.

Reply:
@VictorVelarde @zbigg  
In this case:

```
now, if the app doesn't have ReactIntlProvider, c4r would fallback to en-US
else we would use what user has specified in app
```

If the app doesn't have ReactIntlProvider. Will you be unable to change the language?

Reply:
Correct